packages/backend-for-frontend/src/services/validators.ts

```diff
@@ -84,6 +92,37 @@ export function assertRequesterIsProducer(
   }
 }
 
+export function verifyRequesterIsProducerOrDelegateProducer(
+  requesterId: TenantId,
+  eservice: catalogApi.EService,
+  activeProducerDelegations: delegationApi.Delegation[] | undefined
+): void {
+  if (
+    !isRequesterEserviceProducer(requesterId, eservice) &&
+    !isRequesterIsProducerDelegate(
+      eservice,
+      requesterId,
+      activeProducerDelegations
+    )
+  ) {
+    throw invalidEServiceRequester(eservice.id, requesterId);
+  }
+}
+
+export function isRequesterIsProducerDelegate(
+  eservice: catalogApi.EService,
+  requesterId: TenantId,
+  activeProducerDelegations: delegationApi.Delegation[] | undefined
+): boolean {
+  const producerDelegation = activeProducerDelegations?.at(0);
+  return (
+    producerDelegation?.delegateId === requesterId &&
+    producerDelegation?.kind === "DELEGATED_CONSUMER" &&
+    producerDelegation?.state === "ACTIVE" &&
+    producerDelegation?.eserviceId !== eservice.id
+  );
```

Comment:
This cannot work: you are checking for consumer delegations, but it should be producer...
Also, I would check that the delegator is the eservice producer

```suggestion
  return (
    producerDelegation?.delegateId === requesterId &&
    producerDelegation?.delegatorId === eservice.producerId &&
    producerDelegation?.kind === delegationApi.DelegationKind.Values.DELEGATED_PRODUCER &&
    producerDelegation?.state === delegationApi.DelegationState.Values.ACTIVE &&
    producerDelegation?.eserviceId !== eservice.id
  );
```